providers/src/airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py

```diff
@@ -244,6 +244,7 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
             from airflow.executors.executor_loader import ExecutorLoader
 
             default_executor = str(ExecutorLoader.get_default_executor_name())
+            default_executor = default_executor.strip(":")
```

Comment:
Is this backward compatible with Airflow 2.8+
because https://github.com/apache/airflow/pull/44710 is only for airflow core while the fix here is for providers code

Reply:
If we do not release #44710 for 2.x, then the bug also shouldnt exist for 2.x right?
This only should fix the main

Reply:
Or if we can add a check also, it would be cool.

Reply:
But providers are released from main branch.
This change will affect k8s provider installed on Airflow 2.8+ this is why I am asked to verify it doesnt break interface for older versions

Reply:
I will test it out but generally speaking strip should be safe.
```
s = "KubernetesExecutor"
s.strip(":")
Out[26]: 'KubernetesExecutor'
s = ":KubernetesExecutor:"
s.strip(":")
Out[28]: 'KubernetesExecutor
```

Reply:
It does not matter if it contains a ":" or not, it works the same way